src/imperative/imperative.cc

```diff
@@ -350,6 +350,15 @@ nnvm::Symbol Imperative::GetDeferredComputeSymbol(const std::vector<NDArray *> &
         << "must have a deferred compute history associated with them.";
     s.outputs.emplace_back(ndoutput->deferredcompute_entry_);
   }
+  nnvm::DFSVisit(s.outputs, [&](const nnvm::ObjectPtr& n) {
```

Comment:
I think this needs to run on a copy of the graph, because in general there can still be a call to `Compute` after `GetDeferredComputeSymbol`.